src/runtime_src/core/tools/xbutil2/OO_MemWrite.cpp

```diff
@@ -58,16 +58,16 @@ OO_MemWrite::execute(const SubCmdOptions& _options) const
 {
   XBU::verbose("SubCommand option: write mem");
 
-  // Parse sub-command ...
```

Comment:
Could you elaborate why the following changes were required.   These changes bind's this code to the "--help" option name and no longer relies on the "m_help" variable.

Looking over the comments it is mentioned that:
_In addition, a bug where --write-mem's help menu returned a non-zero exit code has been fixed._

I assume that this is the bug that has been addressed, but not 100% sure.

Reply:
Yes, that is the bug! `process_arguments` was originally throwing an error when using `xbutil advanced --write-mem --help` due to the missing `--address` option. I then tried to make the help option check consistent with OO_MemRead.cpp. From my understanding, "m_help" is not set until `process_arguments` is called.